code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -30,28 +30,39 @@
 		return
 	if(!cargo_holder)
 		return
-	if(istype(target,/obj))
+	if(istype(target, /obj))
 		var/obj/O = target
-		if(!O.anchored)
-			if(cargo_holder.cargo.len < cargo_holder.cargo_capacity)
-				chassis.visible_message("[chassis] lifts [target] and starts to load it into cargo compartment.")
-				O.anchored = 1
-				if(do_after_cooldown(target))
-					cargo_holder.cargo += O
-					O.loc = chassis
-					O.anchored = 0
-					occupant_message("<span class='notice'>[target] successfully loaded.</span>")
-					log_message("Loaded [O]. Cargo compartment capacity: [cargo_holder.cargo_capacity - cargo_holder.cargo.len]")
-				else
-					O.anchored = initial(O.anchored)
-			else
-				occupant_message("<span class='warning'>Not enough room in cargo compartment!</span>")
-		else
+		if(istype(target, /obj/machinery/power/supermatter_crystal)) //No, you can't pick up the SM with this you moron, did you think you were clever?
+			var/obj/machinery/power/supermatter_crystal/SM = target
+			var/obj/mecha/working/ripley/R = chassis
+			if(R.cargo)
+				R.cargo.Cut() //We don't want to drop cargo that just spam hits the SM, let's delete it
+			occupant_message("<span class='userdanger'>You realise in horror what you have done as [chassis] starts warping around you!</span>")
+			chassis.occupant.dust()
+			SM.Bumped(chassis)
```

Comment:
You shouldn't need to create the `SM` variable. Just call `target.Bumped(chassis)`

Reply:
It gets confused when I do this and claims `Bumped()` isn't defined, have I missed something silly?

Reply:
I might be the silly one here, I thought `Bumped()` is defined on the atom level. If it's not then ignore this review. If it is.. it should work though.